ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -59,121 +63,207 @@ public CCodeGenerator() {
         config.setClassForTemplateLoading(CCodeGenerator.class, "/templates");
     }
 
-    public Path getRequestOutputPath(final Request request) {
-        return Paths.get(outputDirectory + "/ds3_c_sdk/src/requests/" + RequestHelper.getNameRootUnderscores(request.getName()) + ".c");
+    public void setFileUtils(final FileUtils fileUtils) {
```

Comment:
To be able to test specific aspects of code generation, instead of calling generate(), which was the only place to set the Spec, FileUtils, and Path.  Generate could be called multiple times with different params, so this isn't really that different.

The other option would be to create a 2nd constructor, or a CCodeGenerator.build


    @Test
    public void testEmbeddedResponseStructParser() throws IOException, ParserException, TypeRenamingConflictException, ResponseTypeNotFoundException, ParseException {
        final TestFileUtilsImpl fileUtils = new TestFileUtilsImpl();

        final Ds3SpecParser parser = new Ds3SpecParserImpl();
        final Ds3ApiSpec spec = parser.getSpec(CCodeGenerator_Test.class.getResourceAsStream("/input/ComplexTypedefStruct.xml"));
        final CCodeGenerator codeGenerator = new CCodeGenerator();
        codeGenerator.setFileUtils(fileUtils);
        codeGenerator.setSpec(spec);
        codeGenerator.generateResponseStructParsers(fileUtils.getOutputStream());